Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1947,11 +1947,11 @@ std::tuple<Ref<WebProcessProxy>, SuspendedPageProxy*, ASCIILiteral> WebProcessPo
 
     // FIXME: We should support process swap when a window has been opened via window.open() without 'noopener'.
     // The issue is that the opener has a handle to the WindowProxy.
-    if (navigation.openedByDOMWithOpener() && !page.preferences().processSwapOnCrossSiteWindowOpenEnabled())
+    if (navigation.openedByDOMWithOpener() && !!page.openerFrame() && !page.preferences().processSwapOnCrossSiteWindowOpenEnabled())
         return { WTFMove(sourceProcess), nullptr, "Browsing context been opened by DOM without 'noopener'"_s };
 
     // FIXME: We should support process swap when a window has opened other windows via window.open.
-    if (navigation.hasOpenedFrames())
+    if (navigation.hasOpenedFrames() && page.hasOpenedPage() && !page.preferences().processSwapOnCrossSiteWindowOpenEnabled())
```

Comment:
Why do we still check the (untrusted) information sent from the WebProcess? Shouldn't we drop it now that we get it from the UIProcess?

Reply:
The WebPageProxy's m_openedPages does not yet account for COEP and noopener (and the value of noopener in WindowFeatures during the createNewPage call doesn't seem to be right when window.open is called from DOM). To cover those latter two cases until that's done, we check both.